src/Services/QueuedJobHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Symbiote\QueuedJobs\Services;
+
+use Monolog\Handler\AbstractProcessingHandler;
+use SilverStripe\Core\Injector\Injectable;
+use Symbiote\QueuedJobs\DataObjects\QueuedJobDescriptor;
+
+/**
+ * Writes log output to a job descriptor
+ */
+class QueuedJobHandler extends AbstractProcessingHandler
+{
+    use Injectable;
+
+    /** @var QueuedJob */
+    protected $job;
+
+    /** @var QueuedJobDescriptor */
+    protected $jobDescriptor;
+
+    public function __construct(QueuedJob $job, QueuedJobDescriptor $jobDescriptor)
+    {
+        $this->job = $job;
+        $this->jobDescriptor = $jobDescriptor;
+    }
+
+    /**
+     * @return QueuedJob
+     */
+    public function getJob(): QueuedJob
```

Comment:
yeah, that would be cool, but we still have to support php 5.6 for some while :)